edx_rest_api_client/auth.py

```diff
@@ -48,3 +48,16 @@ def __call__(self, r):
         """ Update the request headers. """
         r.headers['Authorization'] = 'Bearer {}'.format(self.token)
         return r
+
+
+class OidcIdTokenAuth(AuthBase):
```

Comment:
`JwtAuth` generates a JWT with each request to avoid timeout issues. If we were to only generate it once during instantiation, you run the risk of making API calls with an expired token.